server/item/stack.go

```diff
@@ -224,6 +224,9 @@ func (s Stack) Value(key string) (val any, ok bool) {
 // WithEnchantments returns the current stack with the passed enchantments. If an enchantment is not compatible
 // with the item stack, it will not be applied.
 func (s Stack) WithEnchantments(enchants ...Enchantment) Stack {
+	if _, ok := s.item.(Book); ok {
+		s.item = EnchantedBook{}
```

Comment:
Wouldn't usually agree with this but I think with enchanted books this is fair enough and makes sense.